### PR TITLE
scripts/zynq7000: move flash image creation to build process

### PR DESCRIPTION
**Pull request by HBuczynski**
## Description

 - fix file checking condition
 - move flash image creation to build process
 
JIRA: PD-203

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## How Has This Been Tested?

- [ ] Already covered by automatic testing.
- [ ] New test added: (add PR link here).
- [x] Tested by hand on: (qemu-system-aarch64).

## Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing linter checks and tests passed.
- [ ] My changes generate no new compilation warnings for any of the targets.

## Special treatment

- [ ] This PR needs additional PRs to work (list the PRs, preferably in merge-order).
- [ ] I will merge this PR by myself when appropriate.

### CODE REVIEW

File: scripts/armv7a9-zynq7000.sh [hunk 1]
```diff
@@ -7,19 +7,15 @@
...
-	echo "Missing required files, check $IMG_PLO_ZYNQ7000, $IMG_PHOENIX_ZYNQ7000 and $DTB_ZYNQ7000"
-	exit 1
-fi
-
-# Create file as counterpart of NOR flash memory
-rm -f "$IMG_FLASH_QEMU"
-dd if="$IMG_PHOENIX_ZYNQ7000" of="$IMG_FLASH_QEMU" bs=4M
-truncate -s 16M "$IMG_FLASH_QEMU"
+for FILE in "$IMG_PLO_ZYNQ7000" "$IMG_FLASH_QEMU" "$DTB_ZYNQ7000"; do
+    if [ ! -f "$FILE" ]; then
```
Comment (nalajcie) [inline scripts/armv7a9-zynq7000.sh:14]: You probably wanted to use `\t` here

File: _targets/build.project.armv7a9-zynq7000 [hunk 2]
```diff
@@ -78,14 +78,24 @@ b_build_target() {
...
+	cp "${PREFIX_PROG_STRIPPED}phoenix-${TARGET_FAMILY}-${TARGET_SUBFAMILY}.elf" "${PREFIX_BOOT}/"
+}
+
+
+b_flash_image() {
+	local flash_path="${PREFIX_BOOT}/flash-armv7a9-zynq7000.bin"
+
+	rm -f "${flash_path}"
+	dd if="${PREFIX_BOOT}/phoenix-armv7a9-zynq7000.disk" of="${flash_path}" bs=4M 2>/dev/null
+	truncate -s 16M "${flash_path}" 2>/dev/null
```
Comment (nalajcie) [inline _targets/build.project.armv7a9-zynq7000:91]: please do note that "empty" flash should be filled with `0xFF` and not `0x00`. I assume this doesn't matter for now, but in the future the correct approach should be something like
```bash
dd if=/dev/zero bs=4M count=4 2>/dev/null | tr '\000' '\377' > "${flash_path}"
```

and then overwriting parts of it with desired contents with `conv=notrunc`

**Review by nalajcie — COMMENTED** [review]
spaces to tabs?

File: scripts/armv7a9-zynq7000.sh (hunk 1 above)
Comment (HBuczynski) [reply scripts/armv7a9-zynq7000.sh]: Done

File: _targets/build.project.armv7a9-zynq7000 (hunk 2 above)
Comment (HBuczynski) [reply _targets/build.project.armv7a9-zynq7000:91]: If the future modification is needed, I will take yours proposition into account. Currently, I will leave it as it is.

**Review by nalajcie — APPROVED** [review]

**Pull request merged**